numpy/core/tests/test_multiarray.py

```diff
@@ -4635,6 +4643,20 @@ def test_roundtrip_filename(self):
         y = np.fromfile(self.filename, dtype=self.dtype)
         assert_array_equal(y, self.x.flat)
 
+    @pytest.mark.skipif(pathlib is None, reason="pathlib not found")
+    def test_roundtrip_pathlib(self):
+        p = pathlib.Path(self.filename)
+        self.x.tofile(p)
+        y = np.fromfile(p, dtype=self.dtype)
+        assert_array_equal(y, self.x.flat)
+
+    @pytest.mark.skipif(pathlib is None, reason="pathlib not found")
+    def test_roundtrip_dump_pathlib(self):
+        p = pathlib.Path(self.filename)
+        self.x.dump(p)
+        y = np.load(p, allow_pickle=True)
+        assert_array_equal(y, self.x)
+
```

Comment:
`self.x.dump(p)` is failing to write to the file on PyPy, while `self.x.tofile(p)` is working fine. Is there a call to `flush` missing somewhere, or a missing `close` to trigger the write?

Reply:
Good catch, #13684.